tests/e2e/macvtap_test.go

```diff
@@ -0,0 +1,301 @@
+
+				nodes, _ := clientset.CoreV1().Nodes().List(metav1.ListOptions{})
+				for _, node := range nodes.Items {
+					expectedQuantity, err := resource.ParseQuantity(strconv.Itoa(quantity))
+					Expect(err).NotTo(HaveOccurred())
+
+					// confirm capacity is OK
+					macvtapCapacity := node.Status.Capacity[v1.ResourceName(expectedResourceName)]
+					Expect(macvtapCapacity).To(Equal(expectedQuantity))
+
+					// confirm allocatable is OK
+					macvtapAllocatable := node.Status.Allocatable[v1.ResourceName(expectedResourceName)]
+					Expect(macvtapAllocatable).To(Equal(expectedQuantity))
+				}
+			})
+		})
+	})
+
+	Describe("macvtap resource creation", func() {
+
+		Context("WHEN a macvtap interface is configured as a secondary interface", func() {
+			macvtapIfaceName := "macvtap0"
+			postUrl := "/apis/k8s.cni.cncf.io/v1/namespaces/%s/network-attachment-definitions/%s"
+			nad := `
+					{
+						"apiVersion":"k8s.cni.cncf.io/v1",
+						"kind":"NetworkAttachmentDefinition",
+						"metadata": {
+							"name":"%s",
+							"namespace":"%s",
+							"annotations": {
+								"k8s.v1.cni.cncf.io/resourceName": "%s"
+							}
+						},
+						"spec":{
+							"config":"{ \"cniVersion\": \"0.3.1\", \"name\": \"%s\", \"type\": \"macvtap\"}"
+						}
+					}
+				`
+
+			BeforeEach(func() {
+				clientset.RESTClient().
+					Post().
+					RequestURI(fmt.Sprintf(postUrl, namespace, macvtapIfaceName)).
+					Body([]byte(fmt.Sprintf(nad, macvtapIfaceName, namespace, getMacvtapResourceName(lowerDevice), macvtapIfaceName))).
+					Do()
+			})
+
+			AfterEach(func() {
+				clientset.RESTClient().
+					Delete().
+					RequestURI(fmt.Sprintf(postUrl, namespace, macvtapIfaceName)).
+					Do()
+			})
+
+			Context("WHEN a pod requests aforementioned macvtap resource", func() {
+				podName := "megapod"
+				containerName := "tinycontainer"
+
+				container := v1.Container{
+					Name:    containerName,
+					Image:   "alpine",
+					Command: []string{"/bin/sh", "-c", "sleep 999999"},
+					Resources: v1.ResourceRequirements{
+						Limits: getMacvtapResourceRequest(lowerDevice, 1),
+					},
+				}
+
+				pod1 := &v1.Pod{
+					TypeMeta: metav1.TypeMeta{
+						Kind:       "Pod",
+						APIVersion: "v1",
+					},
+					ObjectMeta: metav1.ObjectMeta{
+						Name:        podName,
+						Namespace:   namespace,
+						Annotations: getMacvtapNetworkAnnotations(macvtapIfaceName),
+					},
+					Spec: v1.PodSpec{
+						Containers: []v1.Container{container},
+					},
+				}
+
+				BeforeEach(func() {
+					_, err := clientset.CoreV1().Pods(namespace).Create(pod1)
+					Expect(err).NotTo(HaveOccurred())
+
+					// wait for pod to be ready
+					waitForPodReadiness(podName, namespace, Timeout)
+				})
+
+				AfterEach(func() {
+					err := clientset.CoreV1().Pods(namespace).Delete(podName, nil)
+					Expect(err).NotTo(HaveOccurred())
+				})
+
+				It("THEN the pod successfully gets a second interface, of macvtap type, with configured MAC address", func() {
```

Comment:
With `It` we usually use "should". So you can read it "It should"

Reply:
Done.